include/qore/intern/qore_socket_private.h

```diff
@@ -989,12 +1001,12 @@ struct qore_socket_private {
 #else
 	    int rc = asyncIoWait(timeout_ms, false, true, "Socket", "connectINETTimeout", xsink);
 #endif
-	    if (xsink && *xsink)
+	    if (*xsink)
                return -1;
 
 	    //printd(5, "asyncIoWait(%d) returned %d\n", timeout_ms, rc);
 	    if (rc == QORE_SOCKET_ERROR && sock_get_error() != EINTR) {
-	       if (xsink && !only_timeout)
+	       if (!only_timeout)
 		  qore_socket_error(xsink, "SOCKET-CONNECT-ERROR", "error in asyncIoWait() with Socket::connect() with timeout", 0, 0, 0, ai_addr);
```

Comment:
same as above - `qore_socket_error()` has the check so it will be easier to remove later